mhs/inbound/inbound/request/handler.py

```diff
@@ -1,46 +1,70 @@
-
 import tornado.web
 
+from mhs_common.state import persistence_adaptor as pa
+from mhs_common.state import work_description as wd
+from mhs_common.state.persistence_adaptor import PersistenceAdaptor
 import mhs_common.messages.ebxml_ack_envelope as ebxml_ack_envelope
 import mhs_common.messages.ebxml_envelope as ebxml_envelope
 import mhs_common.messages.ebxml_request_envelope as ebxml_request_envelope
+from utilities import message_utilities
+import mhs_common.workflow as workflow
 
+logger = log.IntegrationAdaptorsLogger('INBOUND_HANDLER')
 
 
 class InboundHandler(tornado.web.RequestHandler):
     """A Tornado request handler intended to handle incoming HTTP requests from a remote MHS."""
+    state_store: PersistenceAdaptor
+    party_id: str
+    workflows: Dict[str, workflow.CommonWorkflow]
 
-    def initialize(self, callbacks: Dict[str, Callable[[str], None]], party_id: str):
+    def initialize(self, workflows: Dict[str, workflow.CommonWorkflow],
+                   state_store: pa.PersistenceAdaptor, party_id: str):
         """Initialise this request handler with the provided dependencies.
-
-        :param callbacks: The dictionary of callbacks to use when a message is received.
+        :param workflows:
+        :param state_store: The state store
         :param party_id: The party ID of this MHS. Sent in ebXML acknowledgements.
         """
-        self.callbacks = callbacks
+        self.workflows = workflows
         self.party_id = party_id
+        self.state_store = state_store
 
-    def post(self):
-        logging.debug("POST received: %s", self.request)
-        logging.debug("Body: %s", self.request.body)
+    async def post(self):
+        logger.info('001', 'Inbound POST recieved: {request}', {'request': self.request})
 
         request_message = ebxml_request_envelope.EbxmlRequestEnvelope.from_string(self.request.headers,
                                                                                   self.request.body.decode())
-        ref_to_id = request_message.message_dictionary[ebxml_envelope.RECEIVED_MESSAGE_ID]
-        logging.debug("Message received is in reference to '%s'", ref_to_id)
-
-        if ref_to_id in self.callbacks:
+        ref_to_message_id = self.extract_ref_message(request_message)
+
+        logger.info('002', 'Message received has reference to {messageID}', {'messageID': ref_to_message_id})
+
+        try:
+            work_description = await wd.get_work_description_from_store(self.state_store, ref_to_message_id)
+        except wd.EmptyWorkDescriptionError as e:
+            logger.warning('003', 'No work description found in state store with {messageId}',
+                           {'messageId': ref_to_message_id})
+            raise tornado.web.HTTPError(500, 'No work description in state store, unsolicited message '
+                                             'received from spine',
+                                        reason="Unknown message reference") from e
```

Comment:
I'm not sure about the specific workflow that will change this, but yes your understanding is correct, eventually this will be updated to support unsolicited inbound messages